#                                        Python with Pavan

### Pyhton is programming language to tell computer what to do. It is like step by step instructions. It is Simple to read and write (almost like English) and used for many things like designing Websites, analyzing and interpreting data, automating taks, making apps, AI, ML and more. 
### Python was created by Guido van Rossum in 1989 and released in 1991. It wasn't built specifically for data science, later in 20th century Science has involved in programming languages, because of Pyhton's simplicity and flexibility, People started using it in data fields.
### Though R programming is used for Data science, we prefer Python because of its 
1. Huge libraries for data science
* Numpy - for fast numerical calculations
* Pandas - for data manipulation
* Matplotlib and Seaborn - for Visualizations
* Scikit-learn - for Machine learning
* Tensorflow and PyTorch - for Deep Learning

2. Ease of Learing

3. Integration with Other Technologies

4. Strong Community Support

### We'll dive into Data Science in future sessions. For now, let's focus solely on programming concepts without getting into theory.

## Numbers in Python

Types of Numbers
1. Integers - Whole nummbers (10, -5.....)
2. float - Decimal Numbers (10.5, -3.14)
3. Complex - numbers with real and imaginary parts (3+4j)
Let's focus on 
* Integers and Floating numbers
* BASIC ARITHEMATIC
* Difference between classic division and floor division.
* Object in Assignment

### Basic Arithematic

In [5]:
# Addition
print (2+1) # print returns the value (all lower cases.)

3


In [6]:
# Subtraction
print(1000- 53)

947


In [7]:
# Multiplication
print ( 876*543)

475668


In [8]:
# Division
43/2

21.5

In [9]:
# Floor Divison
43//2

21

### WAIT! What just happened?
In step 8, the result is 21.5, but in step 9, it's 21. Why?
That's because we used floor division.
The // operator (double forward slashes) removes the decimal part without rounding and returns an integer result.

### So, what if we only want the remainder after division?
Good question! Let's find out...

In [11]:
# Modulo
43%2

1

In [13]:
# Power
print (2**3) # two asterisk for power

8


In [15]:
# can also do square roots this way
print(4 ** (1/2)) # Nothing but 4**0.5

2.0


In [16]:
# That's Cool but how can we do it for cube root ? Simple,
print(8**(1/3))

2.0


In [17]:
# Order of operations followed in Python (BODMAS Rule)
print(1+10*13-6)

125


### Assigning Variables
We see how can we use the numbers in Python above. Now, we'll look into how can we treat variables as numbers in Pyhton.
We can assign numbers to a variable (Object) by using equal sign (=). suppose a = 9. when I call this Python Script, Python will treat a as a number i.e 9.

*Note - There is difference between = and ==. One equal sign '=' is Assignment Operator (Used to assign a value). and Double equal sign '==' is an Comparision Operator (Used to compare a value). we will talk about this in later sessions.*

In [1]:
# Assigning 5 to variable 'a'
a = 5

In [2]:
print(a)

5


When we call a it will return the value in it.
Also, We can reassign the values. ( we can rewrite the value of 'a')

In [3]:
a= 10
print(a)

10


In [4]:
a+a

20

In [5]:
a -5

5

There are some rules when creating a variable name. It must have:
1. It shouldn't start with number
2. It shouldn't start with Special characters.(!,@,#,$,%...)
3. Instead of Space use '_'
4. Best practice is to name with lowercases
5. Avoid using in built function names like str, list etc.

Best Practice is to name your variables that can be track easily. 

### Let's wrap up this session with a small coding challenge.
I worked for TCS in 2023 and my annual package is 11 lakhs. There will be 10% of tax deductions and 10% of increment on salary every year. How much is my salary after tax cuttings in 2025 ?

In [9]:
# Initial values
my_base_salary = 1100000   # in year 2023
increment_rate = 0.10  # 10% increment
tax_rate = 0.10        # 10% tax

# Year 2024
salary_2024 = my_base_salary * (1 + increment_rate) # see the variable names, can we track these names easily ? 
after_tax_2024 = salary_2024 * (1 - tax_rate) 

# Year 2025
salary_2025 = salary_2024 * (1 + increment_rate) #Yes, We can track variables. 
after_tax_2025 = salary_2025 * (1 - tax_rate)

# Output
print("Salary in 2025 after tax:", round(after_tax_2025, 2))


Salary in 2025 after tax: 1197900.0


In this session, we learned some basics of Numbers and Variables in Python along with some arithematic operations and variable assignments.

We'll catch up in later sessions and be proficient in Pyhton and solve complex problems by the end of course. 

till then....

### PRACTICE.....PRACTICE....PRACTICE